docs/docs/data-fetching.md

```diff
@@ -0,0 +1,169 @@
+
+The simplest way to source data is to take advantage of an existing source plugin. To install the `gatsby-source-graphql`:
+
+```shell
+npm install --save gatsby-source-graphql
+```
+
+Then, add the plugin to your `gatsby-config`:
+
+```javascript:title=gatsby-config.js
+module.exports = {
+  plugins: [
+    {
+      resolve: `gatsby-source-graphql`,
+      options: {
+        typeName: `GitHub`,
+        fieldName: `github`,
+        url: `https://api.github.com/graphql`,
+        headers: {
+          Authorization: `Bearer your-github-token`,
+        },
+      },
+    },
+  ],
+}
+```
+
+> Because the GitHub GraphQL API requires you to be authenticated to make requests, you need to create a token that you would replace in the header for Authorization that says "your-github-token". You can [secure your key using environment variables](/docs/environment-variables/) if you're pushing code to a public repository.
+
+Source plugins take advantage of the [`sourceNodes` API](/docs/node-apis/#sourceNodes) and the [`createNode` action](/docs/actions/#createNode) provided by Gatsby to make your data queryable during the build process.
+
+#### Writing a query to gather the static data needed for a page
+
+With the source plugin installed and added to your config, you can write a [static query](/docs/use-static-query/) that will pull the data from GraphQL API and will retrieve the necessary data while building the site.
+
+```jsx:title=src/pages/index.js
+import React from "react"
+import { graphql, useStaticQuery } from "gatsby" // highlight-line
+
+const IndexPage = () => {
+  // highlight-start
+  const gatsbyRepoData = useStaticQuery(graphql`
+    query {
+      github {
+        repository(name: "gatsby", owner: "gatsbyjs") {
+          id
+          nameWithOwner
+          url
+        }
+      }
+    }
+  `)
+  // highlight-end
+
+  return (
+    <section>
+      <p>
+        Build Time Data: Gatsby repo{` `}
+        <a href="gatsbyRepoData.github.repository.nameWithOwner.url">
+          {gatsbyRepoData.github.repository.nameWithOwner} // highlight-line
+        </a>
+      </p>
+    </section>
+  )
+}
+
+export default IndexPage
+```
+
+### Fetching data at runtime
+
+For fetching data at runtime, you can use any method to retrieve data that you would use in a regular React app.
+
+#### Retrieving data with the `fetch` API
+
+The `fetch` API is a modern implementation of the older, well-supported `XMLHttpRequest`.
+
+With the `useState` and `useEffect` hooks from React, you can query for the data once when the page loads, and save the data returned to a variable called `starsCount`. Every time the page is refreshed, `fetch` will go to the GitHub API to retrieve the most up-to-date version of the data.
+
+```jsx:title=src/pages/index.js
+import React, { useState, useEffect } from "react" // highlight-line
+import { graphql, useStaticQuery } from "gatsby"
+
+const IndexPage = () => {
+  const gatsbyRepoData = useStaticQuery(graphql`
```

Comment:
The build time and runtime data usage here is a little confusing to detangle, especially since the `ClientDataExample` example that follows only includes runtime data after seeing the `useStaticQuery` hook here. Is there a way to isolate these API calls into different examples or label/explain them so it's more clear what's rendered when and why?

Reply:
I added a build time example and reworded it to call the example on page an "adaption". I also added comments to the second section (with build and runtime) to label which is which and added a lengthier explanation below the code to try and explain what was happening. Think that is enough to unpack it or do you have other suggestions?

Reply:
Seems sufficient to me!